### DISCUSSION TITLE
Programmatically Generating & Applying a Prisma Schema

### DISCUSSION THREAD

**Discussion by zacwm** [prisma/prisma#18265] [Q&A] [answered]
Hello!

I've been looking into ways to allow one of my projects (a Discord bot) to have support for many different types of databases that the user can pick from. Prisma appears to be the way to go! 🙌 

However with my project, I have a module like system that I use to enable/disable or change around. With this I wish to be able to possibly specify a Prisma schema just relevant to that module for it to then combine and apply it to be used instantly after it loads and parses all modules.

Is there a way I can apply schemas through a function in the process?

**Comment by BossDaily** [comment]
Im having the same issue, and I am trying to do something similar with discord bots (but with a plugin system)
I've found this so far: https://github.com/valentinpalkovic/prisma-json-schema-generator

**Comment by nurul3101** [comment, marked as answer]
Hello @zacimac 👋 

If I understand correctly, do you want to have multiple prisma schemas and multiple `PrismaClients`, one for each database in your project?

**Reply by zacwm** [reply]
Hey!

With how I have structured my project, I have a modules folder that create more features for the core of the project. So I wish to have a way to either have multiple Prisma schemas, or have some way to parse them all together and apply them as one.

I haven't been able to look too much into Prisma right now, but if having multiple Prisma clients is required for my use case then that's fine.

**Reply by nurul3101** [reply]
I would recommend you to have a look at this GitHub Issue to get an idea on how users are using Multiple Databases with Prisma
- https://github.com/prisma/prisma/issues/2443

Please let us know in case you run into any issues.
